L/LibGit2/build_tarballs.jl

```diff
@@ -31,9 +31,12 @@ if [[ ${target} == *-mingw* ]]; then
         BUILD_FLAGS+=(-DCMAKE_C_FLAGS="-mincoming-stack-boundary=2")
     fi
 elif [[ ${target} == *linux* ]] || [[ ${target} == *freebsd* ]]; then
-    # If we're on Linux or FreeBSD, explicitly ask for mbedTLS instead of OpenSSL
-    BUILD_FLAGS+=(-DUSE_HTTPS=mbedTLS -DSHA1_BACKEND=CollisionDetection -DCMAKE_INSTALL_RPATH="\$ORIGIN")
+    # If we're on Linux or FreeBSD, explicitly ask for mbedTLS instead of OpenSSL.
+    # Disable NTLM because it requires MD4, which is not provided by our build
+    # of MbedTLS, as it's considered unsafe.
+    BUILD_FLAGS+=(-DUSE_HTTPS=mbedTLS -DSHA1_BACKEND=CollisionDetection -DCMAKE_INSTALL_RPATH="\$ORIGIN" -DUSE_NTLMCLIENT=OFF)
```

Comment:
I'm disabling NTLM for Linux and FreeBSD for the reason explained above, should we disable it everywhere for consistency?

Reply:
@aviks pointed out that ntlm might be necessary for some corporate users behind proxy

Reply:
I guess it could make sense to have it on Windows since most users of NTLM are probably there.

Anyway let's make a decision as it's too bad to block this PR just because of this. If we're unsure, we can also disable NTLM on all OSes since it wasn't supported in previous releases: there would be no regression.

Reply:
I think we should enable MD4 and NTLM on all platforms for consistency.  If someone uses NTLM auth then yes, it is likely they are using Windows, but we also want them to be able to use their Linux CI servers to auth in the same way, for instance.

Reply:
OK, I've tried doing that at https://github.com/JuliaPackaging/Yggdrasil/pull/911. MbedTLS appears to have a nonstandard configuration system where you need to edit a header file directly... :-/